src/Psl/Option/Option.php

```diff
@@ -305,4 +306,78 @@ public function equals(mixed $other): bool
     {
         return Comparison\equal($this, $other);
     }
+
+    /**
+     * Combines two `Option` values into a single `Option` containing a tuple of the two inner values.
+     * If either of the `Option`s is `None`, the resulting `Option` will also be `None`.
+     *
+     * @note: If an element is `None`, the corresponding element in the resulting tuple will be `None`.
+     *
+     * @template Tu
+     *
+     * @param Option<Tu> $other The other `Option` to zip with.
+     *
+     * @return (
```

Comment:
`T is never` is still not supported by PHPStan: https://github.com/phpstan/phpstan/issues/9634

Reply:
This is also quite "edgy" in psalm tbh.
See: https://github.com/vimeo/psalm/issues/8487

You can't really trust it to return either never or null.
I'dd say returning a non-conditional `Option<array{T, Tu}>` is the way to go here.
(as we do in all other places as well)

This will be valid cause psalm checks it this way:
https://psalm.dev/r/9715a63a4f
(as we discovered in https://github.com/azjezz/psl/pull/415 )

Reply:
>You can't really trust it to return either never or null.

Not sure about this. As I've been testing, when it _knows_ that you have a `None` value, it's going to be a consistent `Option<none>`. It's covered in 
https://github.com/azjezz/psl/pull/434/files/4eddf3143fe5eb9c72ce665a0748abad17a97cbb#diff-bda3d7e943382d0856488cf5330399ae97aa009d528b8b896df1ce183014dde4

> I'dd say returning a non-conditional Option<array{T, Tu}> is the way to go here.

By doing this, the result of `Option\none()->zip(Option\some(1))` would be `Psl\Option\Option<list{never, 1}>`, which is wrong as far as I understand, it should be `Option<none>`

> (as we do in all other places as well)

Indeed if we keep a conditional return here, I would be missing it in `zipWith()` 🤦‍♂️

Reply:
The thing is that it will mostly never be `never` imo.
Cause `never` means `none()`, and you always want to work with `some(T)` === `option<T>`.
Making psalm think it is all about the `T` and `never` is just if it's not there.

So you'll probably have:

`Option<int>` + `Option<string>` ::zip() ===  `<Option<int, string>>`.

(of which one could actually be `Option<never>` - but it doesn't matter cause psalm knows `never|string` is `string`.

Reply:
>The thing is that it will mostly never be `never` IMO.

I feel the same, knowing beforehand that you are working with a `none` is going to be found on a bit of an absurd code or simple examples, but having the possibility that static analysis thinks that you can have a `Psl\Option\Option<list{never, int}>` triggers my OCD if I can infer the type better beforehand 🤣

The main problem is later with the `unzip` function. If we don't type it a conditional return and we unzip `Psl\Option\Option<list{never, int}>`, we get a `list{Psl\Option\Option<never>, Psl\Option\Option<int>}`, which is not the real behavior too. 

![image](https://github.com/azjezz/psl/assets/1777519/ebc3c098-2848-45f3-86ad-6742642c2426)

Reply:
If it's typed that way, i'dd say it's a userland issue. You can perfectly provide the correct types.
Or it could ad well actually be never. Anyways, I think removing the condition should be fine here